### PR TITLE
python3Packages.aioinflux: init at 0.9.0

### PR DESCRIPTION
**Pull request by liamdiprose**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] Ubuntu 18.04
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jonringer** [comment]
this has your other PR in it, I would just close the other one, and keep this one.

Also, can you please squash your commits so that they just related to the two packages, commit history should look like:
```
pythonPackages.ciso8601: init at <ver>
pythonPackages.aioinflux: init at 0.9.0
```

**Comment by liamdiprose** [comment]
Thanks for the pointers; I've squashed the commits for the two packages. 

Hopefully everything works with rewritten history :disappointed_relieved:

**Comment by jonringer** [comment]
as long as there's no merge conflicts, rebasing isn't necessary

File: pkgs/development/python-modules/aioinflux/default.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+# , cacheSupport ? false   # TODO: Optional Cache Support
+# , aioredis
+# , lz4
+}:
+
+buildPythonPackage rec {
+  pname = "aioinflux";
+  version = "0.9.0";
+
+  disabled = !(pythonAtLeast "3.6");
```
Comment (jonringer) [inline pkgs/development/python-modules/aioinflux/default.nix]: already a function for this :)
```suggestion
  disabled = pythonOlder "3.6");
```

**Comment by liamdiprose** [comment]
I thought that would be the case, thought I'd just check anyway. I only learnt how to rebase 2 hours ago :stuck_out_tongue:

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS
diff LGTM

```
[6 built, 10 copied (176.9 MiB), 31.3 MiB DL]
https://github.com/NixOS/nixpkgs/pull/70840
3 package were build:
python27Packages.ciso8601 python37Packages.aioinflux python37Packages.ciso8601
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build python27Packages.ciso8601 python37Packages.aioinflux python37Packages.ciso8601

File: pkgs/development/python-modules/aioinflux/default.nix (hunk 1 above)
Comment (liamdiprose) [reply pkgs/development/python-modules/aioinflux/default.nix]: Looks like I chose the wrong package to copy from, `pythonOlder` is much clearer. 

Is this an actual commit suggestion, or are you just being helpful? Accepting your commit suggestion will spoil my perfect commit messages.

There's a couple other packages that didn't get the memo:
```sh
$ rg 'disabled = !\(?pythonAtLeast'  -l
aioprocessing/default.nix
jupyter-repo2docker/default.nix
moviepy/default.nix
nest-asyncio/default.nix
aioinflux/default.nix
```
Maybe we could have a separate commit that cleans them up?

Comment (jonringer) [reply pkgs/development/python-modules/aioinflux/default.nix]: no, for nixpkgs, the commits should make sense when you cherry-pick them to another branch. We should consolidate as many package changes into their related commits.

But, if you had one commit that fixed a build, and another that bumped it... then it might make sense to do 2 commits, but in this case as the first addition, it should all be rolled up into one.

to append changes to the last commit:
```
# do changes ...
git add pkgs/
git commit --amend --no-edit
git push <remote> <branch> --force
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build python27Packages.ciso8601 python37Packages.aioinflux python37Packages.ciso8601

**Comment by risicle** [comment]
Two things:

1. The tests for `aioinflux` don't actually run if you inspect the logs. I suspect their `setup.py` wasn't actually hooked up to a `pytest` test-runner. But when you set up a custom `checkPhase` to make them actually run you realize that they almost all use local networking and so won't work in the sandbox. So you may as well disable the tests and remove all those deps. :man_shrugging: 

2. You're pointing at the wrong github repo for `aioinflux`. It's actually at https://github.com/gusutabopb/aioinflux.

**Comment by jonringer** [comment]
good catches @risicle  :)

**Comment by liamdiprose** [comment]
Very good catches indeed. I did notice the 0 tests ran, but have no idea why I thought nothing of it.

`doCheck` is now false, and the test dependencies have been removed.

Out of curiosity, would it be possible to do the network tests in the check stage? Something like the [bittorrent test](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests/bittorrent.nix).

**Comment by risicle** [comment]
I don't know a lot about the nixos tests - @jonringer

**Comment by jonringer** [comment]
I honestly don't know either, but nixos tests allow for you to have many vm's on a virtual network, so they should in theory be able to do cross computer network communication. However, this is probably overkill for just the package, as it would also require nixos modules to configure the VMs initially.

For python packages, you will want to just run the unit tests, this should give a good indication that the package is valid with it's dependencies, without requiring too much setup or compute.

File: pkgs/development/python-modules/aioinflux/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  src = fetchFromGitHub {
+    owner = "gusutabopb";
+    repo = "aioinflux";
+    rev = "v${version}";
+    sha256 = "0cvzkd05i8bzh76m75s7na2gb0kh5msyyz60ajxpj2by9x6qkxmc";
+  };
+
+  propagatedBuildInputs = [ aiohttp pytz ciso8601 ]
+    ++ lib.optional pandasSupport pandas;
```
Comment (jonringer) [inline pkgs/development/python-modules/aioinflux/default.nix]: if it's highly recommended to use pandas, then I would just add this, otherwise leave out the logic to conditionally add. Python packages are weird and it's awkward if two packages have aioinflux as a dependency, then it get's added twice.

Comment (liamdiprose) [reply pkgs/development/python-modules/aioinflux/default.nix]: Pandas is one of the pretty killer features of the package, so I'll remove the condition.

Does the double-add happen when one package has pandasSupport, while the other overrides/disables it? Differing buildInputs would mean differing hashes, so different install paths. I can see how that would be non-trivial to solve.

**Comment by liamdiprose** [comment]
This rebase is blowing up in my face. Not sure how other peoples commits are seeping in...

File: pkgs/development/python-modules/aioinflux/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    owner = "gusutabopb";
+    repo = "aioinflux";
+    rev = "v${version}";
+    sha256 = "0cvzkd05i8bzh76m75s7na2gb0kh5msyyz60ajxpj2by9x6qkxmc";
+  };
+
+  propagatedBuildInputs = [ aiohttp pytz ciso8601 pandas ];
+
+  # Tests require local networking so won't run in the sandbox
+  doCheck = false;
```
Comment (jonringer) [inline pkgs/development/python-modules/aioinflux/default.nix]: please try to do some tests, you can always scope tests using `pytest -k 'not some_test_string'`

Comment (liamdiprose) [reply pkgs/development/python-modules/aioinflux/default.nix]: Yes, I agree. I've looked upstream but haven't found any suitable test filters yet. I might contribute a few tests so we can verify the package requirements are right.

I understand @FRidh isn't in a hurry to merge any python packages until we solve the scalability problem. So this isn't a super high priority for me, but I am still interested in getting it merged at some point. 

Just letting you know I haven't jumped ship!

Comment (jonringer) [reply pkgs/development/python-modules/aioinflux/default.nix]: it's certainly a salability issue for one person, but not as much of an issue for many

Comment (liamdiprose) [reply pkgs/development/python-modules/aioinflux/default.nix]: It's more solved than I thought then. 
I wonder if we could use mypy for a minimal check for all python packages. It can find if there are missing dependencies.

Comment (jonringer) [reply pkgs/development/python-modules/aioinflux/default.nix]: looks like mypy package logic differs from standard python mechanisms and makes assumptions as to where the packages should be located. Probably not compatible with nix

https://mypy.readthedocs.io/en/latest/running_mypy.html#how-imports-are-found

Comment (liamdiprose) [reply pkgs/development/python-modules/aioinflux/default.nix]: Yeah, I've had trouble with the logic, and spent alot of time in that section of the mypy docs. 

I guess nothing beats actually running the code in the interpreter, and any python package worth packaging will have good test coverage anyway.

Comment (jonringer) [reply pkgs/development/python-modules/aioinflux/default.nix]: you can also do `pythonImportsCheck = [ "foo.bar" ];` to essentially run `python -c 'import foo.bar'` 

I use it extensively https://github.com/NixOS/nixpkgs/blob/3140fa89c51233397f496f49014f6b23216667c2/pkgs/tools/admin/azure-cli/default.nix#L168 to ensure that the azure namespace isn't broken by a rogue __init__.py file

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -3778,6 +3778,12 @@
     githubId = 307589;
     name = "Nathaniel Baxter";
   };
+  liamdiprose = {
+    email = "liam@liamdiprose.com";
+    github = "liamdiprose";
+    githubId = 1769386;
+    name = "Liam Diprose";
+  };
```
Comment (jonringer) [inline maintainers/maintainer-list.nix]: this should be a separate commit

**Review by jonringer — COMMENTED** [review]
sorry for the very long delay

File: maintainers/maintainer-list.nix (hunk 4 above)
Comment (liamdiprose) [reply maintainers/maintainer-list.nix]: Roger roger. To be honest I thought I was the slacker :stuck_out_tongue: 

Now that my country is locked down, I see no better option than to get back into open source. I'll be pushing soon.

**Review by jonringer — COMMENTED** [review]
please resolve conflicts as well

**Comment by prusnak** [comment]
@liamdiprose do you still want to get this in?

**Comment by lopsided98** [comment]
I submitted an updated version of this PR: #111849

**Comment by prusnak** [comment]
Superseded by https://github.com/NixOS/nixpkgs/pull/111849
